sdk/core/Azure.Core.TestFramework/src/TestEnvironment.cs

```diff
@@ -449,5 +481,49 @@ internal static bool GlobalDisableAutoRecording
                 return disableAutoRecording || GlobalIsRunningInCI;
             }
         }
+
+        private void BootStrapTestResources()
+        {
+            try
+            {
+                _bootstrappingSemaphore.Wait();
+
+                if (_bootstrappingAttempted)
+                {
+                    return;
+                }
+
+                string path = Path.Combine(TestEnvironment.RepositoryRoot, "eng", "common", "TestResources",
+                    $"New-TestResources-Wrapper.ps1 {_serviceName} " +
+                    $"-basename {Environment.UserName}-{_serviceName}-{Guid.NewGuid().ToString().Substring(0, 4)} ");
```

Comment:
Why?

Reply:
Why default? So the user doesn't have to enter anything and we get a unique base name.

Reply:
Why do we need a unique base name?

Reply:
The error you get when a duplicate name is used is not a good experience. I think we should probably improve the default behavior in New-TestResources.ps1.

Reply:
Can we make the error good?

Reply:
https://github.com/Azure/azure-sdk-tools/pull/1953

Reply:
Isn't the error "The ... already exists, do you want to replace?"

Reply:
No, it is more obscure than that.